lib/Sema/CheckedCSubst.cpp

```diff
@@ -75,18 +76,254 @@ ExprResult Sema::ActOnTypeApplication(ExprResult TypeFunc, SourceLocation Loc,
 
 }
 
+// Type Instantiation
+RecordDecl* Sema::ActOnRecordTypeApplication(RecordDecl *Base, ArrayRef<TypeArgument> TypeArgs) {
+  assert(Base->isGeneric() && "Base decl must be generic in a type application");
+  assert(Base->getCanonicalDecl() == Base && "Base must be canonical decl");
+ 
+  auto &ctx = Base->getASTContext();
+
+  // Unwrap the type arguments from a 'TypeArgument' to the underlying 'Type *'.
+  auto RawArgs = new (ctx) llvm::SmallVector<const Type *, 4>();
+  for (auto TArg : TypeArgs) {
```

Comment:
You can omit the braces for the for loop.

Reply:
I personally like the braces, because they make the scope clearer: c.f.

for (auto TArg : TypeArgs)
  doSomething(); // ok: run with the loop
  doSomethingElse(); // looks like it's run as part of the loop, but isn't